contracts/HATTimelockController.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.6;
+
+
+import "openzeppelin-solidity/contracts/governance/TimelockController.sol";
+import "./HATVaults.sol";
+
+
+contract HATTimelockController is TimelockController {
+    HATVaults public hatVaults;
+
+    constructor(
+        uint256 _minDelay,
+        address[] memory _proposers,
+        address[] memory _executors
+    // solhint-disable-next-line func-visibility, no-empty-blocks
+    ) TimelockController(_minDelay, _proposers, _executors) {
+    }
+
+    function initializeHATVaults(HATVaults _hatVaults) external onlyRole(EXECUTOR_ROLE) {
+        require(address(_hatVaults) != address(0), "HATTimelockController: HATVaults address must not be 0");
+        require(address(hatVaults) == address(0), "HATTimelockController: Cannot update HATVaults address");
+        hatVaults = _hatVaults;
+    }
+    
+    // Whitelisted functions
```

Comment:
I believe it makes more sense sot give all whitelisted functions the PROPOSER_ROLE, but let's discuss as it can also go this way

Reply:
![image](https://user-images.githubusercontent.com/1306173/145887101-4a308f0d-0d2c-451e-b7a6-467dcc04668a.png)